homeassistant/components/roomba/__init__.py

```diff
@@ -1 +1,76 @@
 """The roomba component."""
+import logging
+
+import async_timeout
+from roomba import Roomba, RoombaConnectionError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+
+from .const import CONF_CERT, CONF_CONTINUOUS, CONF_DELAY, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass, config):
+    """Set up the roomba environment."""
+    if DOMAIN not in config:
+        return True
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data={}
+            )
+        )
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set the config entry up."""
+    # Set up roomba platforms with config entry
+    if config_entry.data is None:
+        return False
+
+    if not config_entry.options:
+        hass.config_entries.async_update_entry(
+            config_entry,
+            options={
+                "certificate": config_entry.data[CONF_CERT],
+                "continuous": config_entry.data[CONF_CONTINUOUS],
+                "delay": config_entry.data[CONF_DELAY],
+            },
+        )
+
```

Comment:
I can't see any place where the options flow changes are taking effect.  Apologizes if I missed it

```
    config_entry.add_update_listener(_update_listener)
```

...
```
async def _update_listener(hass: HomeAssistant, entry: ConfigEntry):
   ...
```